### ISSUE TITLE
Deprecate jax and tensorflow

### ISSUE THREAD

**Issue by RobinPicard** [dottxt-ai/outlines#1725]
The tensor adapters module has support for those 2 tensor libraries, but the backend libraries (`outlines-core`, `xgrammar` and `llguidance`) do not support them. As a result, the tensor adapters for jax and tensorflow are never used and the `Transformers` models using them cannot be used without logits processors (so only unstructured generation is supported).

We should delete the unused code in the tensor adapters and add a deprecation warning in the `Transformers` model.
